src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -93,5 +94,56 @@ public virtual bool CanParseValue(object value)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Checks that the provided value is a valid default value.
+        /// This method should not throw an exception.
+        /// </summary>
+        public virtual bool IsValidDefault(object value)
+        {
+            try
+            {
+                return ToAst(value) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Converts a value to an AST representation. This is necessary for introspection queries
+        /// to return the default value for fields of this scalar type. This method may throw an exception
+        /// or return null for a failed conversion.
+        /// </summary>
+        public virtual IValue ToAst(object value)
+        {
+            try
+            {
+                var serialized = Serialize(value);
+                return serialized switch
+                {
+                    bool b => new BooleanValue(b),
```

Comment:
Do all `DateTime`, `Guid` and other types from "old code" fall back to `_` default case?

Reply:
Yes.